tests/test_views.py

```diff
@@ -786,6 +786,14 @@ def test_dataframe_view_index_setting():
     assert a2.obs.index.values.tolist() == ["a", "b"]
 
 
+def test_ellipsis_index(adata, subset_func, matrix_type):
```

Comment:
Should also test `adata[...]` and `adata[..., subset_var_names]`

Reply:
Good call!

Reply:
Oh! And even more tricky:
- `adata[obs, var, ...]`
- `adata[obs, ..., var]`
- `adata[..., obs, var]`

in these cases it’ll expand to nothing actually.

Reply:
Is this in the scope of this issue though?  I would think this is more along the lines of ND AnnData?

Reply:
If we want to support ellipsis indexing, we should support its full semantics, and those are

> like `:` but for each remaining axis.

So if 2 axes are remaining, it expands to `:, :`, and if 0 axes are remaining, it vanishes.

Reply:
Done.  We error on more than one ellipsis as well now with a more helpful message than before